enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,441 @@
+well as support of the feature (CI).
+
+We propose several options in the [Test Plan](#test-plan) section below.
+
+#### Uncertainty about OpenStack's extra networking resources
+
+The OpenShift Installer as well as [cluster-api-provider-openstack][capo]
+currently create a port with the following properties set for every node:
+
+* Security groups
+* Attached to the network/subnet created during the installation
+* Allowed Address pairs set for the VRRP IP addresses defined by the installer
+* (optional) network trunk
+
+Some of these options (e.g. security groups) are only supported for certain ML2
+drivers. Others (attaching to a given tenant network) only work for the
+`neutron` Ironic networking interface but not for the `flat` one.
+
+It is unclear what happens when the installer tries to set any of these in
+environments whehere they aren't supported. If that results in error, we may
+have to detect these situations, pass them to [CAPO][capo] and modify how we
+create the node and port resources.
+
+This can be mitigated by not using [CAPO][capo] for the baremetal nodes.
+Instead, the OpenShift administrator would create them manually, pass the
+correct Ignition configuration and approve the certificate signing requests
+(CSRs) explicitly.
+
+This manual process of adding compute nodes is described in the [OpenStack
+user-provisioned infrastructure (UPI) document][openstack-upi] and should work
+here as well.
+
+## Design Details
+
+### Test Plan
+
+The OpenStack cloud running CI does not currently support Ironic and bare metal
+machines.
+
+Similarly, switching to a different OpenStack provider or building our own
+cluster are longer-term solutions that we cannot rely on.
+
+As this is an issue faced by other OpenStack bare metal-related projects such
+as Ironic and TripleO, we are going to utilise a similar solution: have the CI
+job build out the OpenStack deployment required.
+
+* The CI job will create virtual machines serving as the OpenStack nodes
+* It will also create virtual machines serving as the bare metal nodes
+* It will install OpenStack on top of the OpenStack virtual nodes
+* It will configure the OpenStack to provide bare metal nodes via
+  virtualbmc/sushy-tools
+* It will run the OpenShift installation on top of that OpenStack
+
+### Graduation Criteria
+
+This enhancement will follow standard graduation criteria.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- Community Testing
+- Sufficient time for feedback
+- Downgrade and scale testing are not relevant to this enhancement
+
+### Upgrade / Downgrade Strategy
+
+This proposal should have no effect on existing OpenShift clusters running on
+OpenStack. We do not plan any changes to the VM-based OpenStack deployments.
```

Comment:
What about the new subnet parameter in the machine pool definition?

Reply:
I am removing the sentence "We do not plan any changes to the VM-based OpenStack deployments." and adding a reference to that new necessary property in the "Implementation work"

Reply:
From my tests 2 and 3 work fine. Note that we'll have a mix of BM nodes and VMs in the same subnet in any case since the control plane still runs on VMs.
4 allows deploying the worker nodes in different subnets and I think is enough to cover the use case, so we don't need to introduce a new attribute in the machine pool definition. Day 2 operation FTW!
I'd like to see 1 work.